x-pack/plugins/fleet/server/services/agents/update_agent_tags.ts

```diff
@@ -39,12 +39,23 @@ export async function updateAgentTags(
     }
   } else if ('kuery' in options) {
     const batchSize = options.batchSize ?? SO_SEARCH_LIMIT;
+
+    const filters = [];
+    if (options.kuery !== '') {
+      filters.push(options.kuery);
+    }
+    if (tagsToAdd.length === 1 && tagsToRemove.length === 0) {
```

Comment:
moved the tags filters before we search for agent ids, this is more efficient on retries
e.g. 50k agents are actioned, 1k conflicted
on retry we can just run one batch on the remaining 1k, rather than executing 10k at a time and finding agents in each batch that are not yet updated

Reply:
Because of these changes to make retry more efficient, I had to remove point in time settings from retry of update tags (otherwise I can't filter on updated state of agent tags).
This change has a side effect that the update tags retry logic is applied on new agents as well that were enrolled since the action started and match the filters.